crawler_mcp/core/rag/deduplication.py

```
@@ -582,3 +582,115 @@ def should_use_backwards_compatibility(
+        Args:
+            source_url: Source URL to check for orphaned chunks
+            
+        Returns:
+            List of orphaned chunk IDs
+        """
+        if not self.vector_service:
+            return []
+            
+        try:
+            # Find all chunks for this source
+            existing_chunks = await self.find_existing_chunks(source_url)
```

Ruff (0.12.2)

Local variable `existing_chunks` is assigned to but never used

Remove assignment to unused variable `existing_chunks`

(F841)